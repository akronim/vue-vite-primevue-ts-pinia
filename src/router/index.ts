import { createRouter, createWebHistory, type RouteComponent, type RouteRecordRaw } from 'vue-router'

export const RouteNames = {
  VUE_TEST_ONE: `vueTestOne`,
  VUE_TEST_TWO: `vueTestTwo`
}

export const routes: RouteRecordRaw[] = [
  {
    path: `/demo/vuetest/one`,
    name: RouteNames.VUE_TEST_ONE,
    component: (): Promise<RouteComponent> => import(`@/views/demo/VueTestOne.vue`)
  },
  {
    path: `/demo/vuetest/two`,
    name: RouteNames.VUE_TEST_TWO,
    component: (): Promise<RouteComponent> => import(`@/views/demo/VueTestTwo.vue`)
  }
]

export const router = createRouter({
  routes,
  //history: createWebHistory(import.meta.env.BASE_URL),
  history: createWebHistory()
})