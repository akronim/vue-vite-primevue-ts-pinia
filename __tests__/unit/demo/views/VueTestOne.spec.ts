import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import VueTestOne from '@/views/demo/VueTestOne.vue'
import { createRouter, createWebHistory } from 'vue-router'
import { routes } from '@/router'
import { h } from 'vue'
import { RouterLink } from 'vue-router'
import { RouteNames } from '@/router'
import VueLogo from '@/assets/logo.svg?component'
import ToastService from 'primevue/toastservice'
import PrimeVue from 'primevue/config'

const mockRouter = createRouter({
    history: createWebHistory(),
    routes: [
        { path: `/`, component: h(`div`) },
        ...routes
    ]
})

vi.mock('@/services/demo/demoProductService', () => ({
    getAllProducts: vi.fn()
}))

describe('VueTestOne', () => {
    const options = {
        global: {
            plugins: [PrimeVue, ToastService, mockRouter]
        }
    }

    it('renders VueLogo and RouterLink', () => {
        const wrapper = mount(VueTestOne, options)

        expect(wrapper.findComponent(VueLogo).exists()).toBe(true)

        expect(wrapper.findComponent(RouterLink).exists()).toBe(true)
        expect(wrapper.findComponent(RouterLink).props().to).toEqual({ name: RouteNames.VUE_TEST_TWO })

        const headEl = wrapper.find('.vue-test-one__head')
        expect(headEl.exists()).toBe(true)

        expect(wrapper.findAll('.vue-test-one__head a').length).toBe(1)
    })

    it('snapshot UI testing', () => {
        const wrapper = mount(VueTestOne, options)
        expect(wrapper.text()).toMatchSnapshot()
    })
})