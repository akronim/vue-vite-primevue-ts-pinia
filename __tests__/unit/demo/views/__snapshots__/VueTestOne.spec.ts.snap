// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VueTestOne > snapshot UI testing 1`] = `"Go To TwoNewCodeNameCategoryQuantityActions No Data Found. Â 50 to 0 of 0"`;
